### DISCUSSION TITLE
How to find file in filecoin by cid of ipfs？

### DISCUSSION THREAD

**Discussion by defrontorg** [filecoin-project/community#543] [Developers]
I observed that cid is not the same in ipfs and filecoin.

**Comment by longfeiWan9** [comment]
@defrontorg 

1. storing Files on IPFS and Filecoin is different. On Filecoin, you make storage deals with SP(Storage providers) to store your files. So when you want to retrieve it, you also need to make retrieval deals with SP to get your files with CID and deal ID. 

1. IPFS and Filecoin are two different protocols or networks. If you only store your files on IPFS, you cannot find them on Filecoin by default unless you also make storage deals with SP on Filecoin. 

1. Filecoin and IPFS are both using IPLD for their data structure. So as long as you are using the same algorithm to generate CID for your files, CID will be the same. Just [different versions](https://docs.ipfs.tech/concepts/content-addressing/#cid-versions), v0 and v1 CID.
